client/extensions/zoninator/components/settings/zone/index.jsx

```diff
@@ -34,12 +43,20 @@ class Zone extends Component {
 
 	hideDeleteDialog = () => this.setState( { showDeleteDialog: false } );
 
+	deleteZone = () => this.props.deleteZone( this.props.siteId, this.props.zoneId );
+
+	saveZoneDetails = ( form, data ) => this.props.saveZone( this.props.siteId, this.props.zoneId, form, data );
+
+	saveZoneFeed = ( form, data ) => this.props.saveFeed( this.props.siteId, this.props.zoneId, form, data.posts );
+
 	render() {
-		const { siteId, siteSlug, translate } = this.props;
+		const { feed, siteId, siteSlug, translate, zone, zoneId } = this.props;
 		const { showDeleteDialog } = this.state;
 
 		return (
 			<div>
+				<QueryFeed siteId={ siteId } zoneId={ zoneId } />
```

Comment:
The first few times this is called, `siteId` and `zoneId` could be null, so it doesn't make much sense to try to query for data in that case. Just realized I have to fix this in my code as well.

Reply:
I noticed this is usually taken care of by the `Query...` components themselves - that's also what I did here. Should we still change it ?

Reply:
So apparently this:

> it’s going to be more performant to check because when query components render `null` they still load themselves into memory. so not rendering a component from the parent uses less memory and CPU time than trying to prevent rendering from within a component

Reply:
There's differing opinions here as per https://a8c.slack.com/messages/C02DQP0FP/-slack-calypso. So I'd say do whatever you feel is best.

Reply:
Yeah, I guess it's not such a big deal performance wise but I found it did clean up the code a little bit by moving that logic outside of the component.